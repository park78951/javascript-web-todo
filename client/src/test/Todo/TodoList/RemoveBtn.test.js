import React from 'react';
import { 
  render, 
  fireEvent, 
} from '@testing-library/react';
import RemoveBtn from '../../../components/Todo/TodoList/RemoveBtn';

describe('<RemoveBtn />', () => {
  it('matches snapshots', () => {
    const { container } = render(<RemoveBtn />);
    expect(container).toMatchSnapshot();
  });

  it('should be clicked', () => {
    const mockRemoveHandler = jest.fn();
    const { getByText } = render(<RemoveBtn removeHandler={ mockRemoveHandler } />);
    const removeBtn = getByText('X');
    fireEvent.click(removeBtn);
    expect(mockRemoveHandler).toHaveBeenCalledTimes(1);
    fireEvent.click(removeBtn);
    expect(mockRemoveHandler).toHaveBeenCalledTimes(2);
  });
});