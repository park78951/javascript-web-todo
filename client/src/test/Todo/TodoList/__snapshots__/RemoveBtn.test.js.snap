// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemoveBtn /> matches snapshots 1`] = `
<div>
  <button
    class="RemoveBtn__Button-w4yd9g-0 hxawzU"
  >
    X
  </button>
</div>
`;
